### PR TITLE
Case 20322, Case 12279, Case 15803: Improve PolyLines to support glow and faceCamera

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/20322/Modify-PolyLines-to-support-everything-Line3D-Overlays-can-do
https://highfidelity.manuscript.com/f/cases/12279/PolyLine-s-registrationPoint-value-is-wrong
https://highfidelity.manuscript.com/f/cases/15803/Crash-while-rendering-PolyLineEntity

New PolyLine properties and re-working how we render polylines to be more efficient.

Test plan:
- run the content/entity/polyline tests from this branch: https://github.com/highfidelity/hifi_tests/tree/f92b377c5a36a484f7a8f601fd9acf2a181e68f6/tests/content/entity/polyline
- verify that the Fingerpaint (beta) app in the marketplace still works as in master
- Open Create and select one of the PolyLines you've created.  Change its registration point.  The line/bounding box should move as expected.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15989/HighFidelity-Beta-PR14586-1960537.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15989/HighFidelity-Beta-PR14586-1960537.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15989/HighFidelity-Beta-Interface-PR14586-1960537.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15996/HighFidelity-Beta-PR14586-8216cec.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15996/HighFidelity-Beta-Interface-PR14586-8216cec.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15996/HighFidelity-Beta-PR14586-8216cec.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15996/HighFidelity-Beta-Interface-PR14586-8216cec.exe

**Comment by qlilaseguinolaza** [comment]
Currently testing

**Comment by qlilaseguinolaza** [comment]
Stopped testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16077/HighFidelity-Beta-PR14586-05e0fb4.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16077/HighFidelity-Beta-PR14586-05e0fb4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16077/HighFidelity-Beta-Interface-PR14586-05e0fb4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16077/HighFidelity-Beta-PR14586-05e0fb4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16077/HighFidelity-Beta-Interface-PR14586-05e0fb4.exe

**Comment by qaAdamSmith82** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16146/HighFidelity-Beta-PR14586-05e0fb4.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16146/HighFidelity-Beta-PR14586-05e0fb4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16146/HighFidelity-Beta-Interface-PR14586-05e0fb4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16146/HighFidelity-Beta-PR14586-05e0fb4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16146/HighFidelity-Beta-Interface-PR14586-05e0fb4.exe

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16234/HighFidelity-Beta-PR14586-0e6eecc.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16234/HighFidelity-Beta-PR14586-0e6eecc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16234/HighFidelity-Beta-Interface-PR14586-0e6eecc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16234/HighFidelity-Beta-PR14586-0e6eecc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16234/HighFidelity-Beta-Interface-PR14586-0e6eecc.exe

**Comment by qlihifitest** [comment]
Currently testing

File: libraries/entities-renderer/src/RenderablePolyLineEntityItem.cpp [hunk 1]
```diff
@@ -19,274 +19,241 @@
...
-    // TODO: adapt polyline into a triangles mesh...
-    return EntityRenderer::getScriptableModel();
+void PolyLineEntityRenderer::updateData() {
+    PolylineData data { glm::vec2(_faceCamera, _glow), glm::vec2(0.0f) };
+    _polylineDataBuffer->setSubData(0, data);
 }
 
 void PolyLineEntityRenderer::doRender(RenderArgs* args) {
-    if (_empty) {
+    if (_numVertices < 2) {
```
Comment (sethalves) [inline libraries/entities-renderer/src/RenderablePolyLineEntityItem.cpp]: if the first (implied?) vertex is 0,0,0 should this be < 1 ?

**Comment by qlihifitest** [comment]
Pausing testing RE changes requested

**Comment by qliacoombs** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16243/HighFidelity-Beta-PR14586-0e6eecc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16243/HighFidelity-Beta-Interface-PR14586-0e6eecc.exe

**Comment by qlihifitest** [comment]
Resuming testing

**Comment by qlihifitest** [comment]
FAIL
Ran content/entity/polyline tests
https://github.com/highfidelity/hifi_tests/tree/f92b377c5a36a484f7a8f601fd9acf2a181e68f6/tests/content/entity/polyline

- Color test FAIL (lines rendered but no color)
- faceCamera 0 FAIL (lines rendered but no color)
- faceCamera 1 FAIL (lines rendered but no color)
- glow - FAIL (lines rendered but no color)
- isUVModeStretch - FAIL (lines rendered but no color)
- linePoints - FAIL (lines rendered but no color)
- normals - FAIL (lines rendered but no color)
- strokeColors - FAIL (lines rendered but no color)
- textures - FAIL (lines rendered, one object with color, one object no color)

Fingerpaint app works as in master

Successfully opened Polyline and changed registration point.

[PR14586 test FAIL 01042019.zip](https://github.com/highfidelity/hifi/files/2728681/PR14586.test.FAIL.01042019.zip)
[TestResults--2019-01-04_13-51-03(local)[2060-DESKTOP-PC].zip](https://github.com/highfidelity/hifi/files/2728682/TestResults--2019-01-04_13-51-03.local.2060-DESKTOP-PC.zip)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16287/HighFidelity-Beta-PR14586-5cded69.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16287/HighFidelity-Beta-PR14586-5cded69.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16287/HighFidelity-Beta-Interface-PR14586-5cded69.exe

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Pull request merged**